Code/Framework/AzCore/AzCore/Name/Name.cpp

```diff
@@ -123,6 +134,32 @@ namespace AZ
             *this = Name();
         }
     }
+
+    void Name::SetNameLiteral(AZStd::string_view name)
+    {
+        if (name[0] != '\0')
+        {
+            m_view = name;
+            if (!NameDictionary::IsReady(false))
+            {
+                // Link ourselves into the deferred list if we're not already in there
+                if (!m_supportsDeferredLoad)
+                {
+                    LinkStaticName(&s_staticNameBegin);
+                }
+            }
+            else
+            {
+                NameDictionary::Instance().LoadDeferredNames(this);
+            }
+
+            m_supportsDeferredLoad = true;
```

Comment:
The other flag is for when the Name is specifically in the name dictionary's linked list, which will only be set when the name is loaded into the dictionary. Depending on the module this name is in, it may not be linked into the dictionary's list of static names yet, so we set that later